esp/esp/program/controllers/lottery.py

```diff
@@ -68,6 +69,16 @@ def __init__(self, program, **kwargs):
         
 
         self.program = program
+
+        if Tag.getProgramTag('program_size_by_grade', self.program):
+            # TODO(benkraft): Consider implementing this.  Stanford's use case
+            # as of Fall 2015 (for which program_size_by_grade was written)
+            # doesn't need it, but we might want to implement it anyway, or
+            # remove the program size logic from the lottery entirely.
+            print ("WARNING: The lottery doesn't support the "
+                   "program_size_by_grade Tag yet.  It will run based on "
+                   "`Program.program_size_max`, which is %s." %
+                   self.program.program_size_max)
```

Comment:
Shouldn't we ignore `Program.program_size_max` if `program_size_by_grade` is set? Since the Tag takes precedence over the the model value, if admins accidentally set `Program.program_size_max` (e.g. during program creation), but then later set the tag, then they will run through all of lottery studentreg as if `Program.program_size_max` was unset. But then suddenly we'll get to the lottery and it'll act as if it's set. If `Program.program_size_max` is accidentally smaller than the sum of the capacities in the Tag, then this will lead to students getting kicked out.

And/or we can print the `program_max_size`, the Tag, and the total number of students registered, and prompt the user if they want to continue with the `program_max_size`, ignore it, or quit to fix things up.

Although this isn't a problem with your code review, this made me realize that we're not handling OverrideFull correctly here. It is valid to have more than `program_max_size` students, but the lottery will strictly cap it there. Should we fix this now, or separately?